### DISCUSSION TITLE
Ammonite / Scala 3

### DISCUSSION THREAD

**Discussion by danbills** [scalameta/nvim-metals#301] [Q&A] [answered]
I've happily got Ammonite scripts working w/ the fix described here:

https://github.com/scalameta/nvim-metals/discussions/281

So now I'm getting greedy and trying to use scala 3 :).

I'm still seeing the issue if I use

```
// scala 3.0.2

LSP[metals] Error fetching Ammonite 2.3.8-124-2da846d2 for scala 3.0.2
```

as well as 

```
// ammonite 2.5.0
// scala 3.0.2

LSP[metals] Error fetching Ammonite 2.5.0 for scala 3.0.2
```

and this just doesn't seem to do anything

```
// ammonite 2.5.0

LSP[metals][Log] No build tool
```

I'm poking around this repo and ammonite-runner but I don't see a mapping to update or where version `2.3.8-...` is getting derived (this version of ammonite doesn't exist, latest if 2.5.0.

I'm happy to scratch my own itch here as you've already been very helpful but a pointer in the right direction is needed.  Thanks again for this great work!

**Comment by ckipp01** [comment, marked as answer]
Hey! Yes, ammonite-runner and Scala 3 is a bit of a different beast. There were a lot of breaking changes there that aren't yet accounted for in ammonite-runner mainly because some artifacts are published by 2.13 and then used in 3. I actually did some initial work on that here https://github.com/alexarchambault/ammonite-runner/pull/127 which you could play around with. Until there is support for Scala 3 in ammonite-runner there isn't a lot we can do in Metals.
